### PR TITLE
Remove remote settings fallback

### PR DESCRIPTION
**Pull request by jkakavas**
Fallback settings were introduced in #33413 where the
`search.remote.*` settings were generalized to `cluster.remote.*`.
This commit removes both the fallback settings and the upgraders
that were introduced in #33537

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/transport/RemoteClusterServiceTests.java [hunk 1]
```diff
@@ -163,33 +161,6 @@ public void testBuildRemoteClustersDynamicConfig() throws Exception {
         assertEquals("quux-proxy:19300", map.get("quux").v1());
         assertEquals(quux.getVersion(), Version.CURRENT.minimumCompatibilityVersion());
 
-        assertSettingDeprecationsAndWarnings(new String[]{"search.remote.quux.seeds", "search.remote.quux.proxy"});
-    }
-
-    public void testBuildRemoteClustersDynamicConfigWithDuplicate() {
```
Comment (jkakavas) [inline server/src/test/java/org/elasticsearch/transport/RemoteClusterServiceTests.java]: I think these were not needed any more now that we don't have both `cluster.remote` and `search.remote` settings to account for but we can reintroduce duplication validation if need be in `RemoteClusterAware.java`

Comment (javanna) [reply server/src/test/java/org/elasticsearch/transport/RemoteClusterServiceTests.java]: correct, we don't need this anymore.

**Review by javanna — APPROVED** [review]
LGTM thanks @jkakavas !

**Review by martijnvg — APPROVED** [review]
:+1:

**Pull request closed without merging**